pkg/apis/resource/types.go

```diff
@@ -442,8 +442,12 @@ type DeviceRequest struct {
 	// all ordinary claims to the device with respect to access modes and
 	// any resource allocations.
 	//
+	// This is an alpha field and requires enabling the DRAAdminAccess
+	// feature gate.
+	//
 	// +optional
 	// +default=false
+	// +featureGate=DRAAdminAccess
 	AdminAccess bool
```

Comment:
How does it work for this to be optional but not a pointer?

Reply:
We have done a few times in the API, by treating "false" the same as "not set". That's okay as long as we have no-one who needs to distinguish between the two, which is the case here.

Reply:
I've changed it because it's consistent with usual practice and thus easier to review. However, I don't think it makes a difference and is worse in one regard.

@LionelJouin, @aojea: you may have to reconsider this in the device status PR because you also have a feature gate.

Here, it doesn't make a difference for users because for them `adminAccess: false` and `adminAccess: null` are identical (= admin access off). They are probably not going to set it to false explicitly. It *would* make a difference if the field was mutable because then a user could set `adminAccess: false` and then because of the "allow setting disabled field if in use" rule change it later when the feature gate is turned off. But the claim spec and allocation status are immutable, so this special case does not apply here.

It is (slightly) worse because in the allocation status, "not set" in the first commit meant "look it up in the spec" to support upgrades from 1.31 where that status field didn't exist yet. Now "not set" means "off", the same as "false". This is entirely theoretical because there is no DRA driver which supported admin access with 1.31 and therefore probably no claim that could trigger this corner case, so it's okay to just add the status field and treat it as if we were adding this feature anew.

@thockin: I had to rebase, so those changes are in a new commit at the end in case that you want to do an incremental review.